CONTRIBUTING.md

```diff
@@ -87,14 +87,73 @@ We use `mkdocs` to build the documentation and a plugin called `keras-autodoc` t
     python auto_doc.py
     ```
 
+##### Option 1: Build only current version of docs
+
 4. Either build the docs, or serve them dynamically:
 
+    Note: Links and pictures might not resolve properly later on when checking with this build.
+
     ```bash
     mkdocs build
     # or
     mkdocs serve
     ```
 
+##### Option 2 (Preferred): Build multi-version doc with `mike`
+
+4. For this you can either checkout and make a local copy of the upstream/gh-pages branch to get the current state of the docs, or just build the branch you are updating:
+
+    Building *one* branch:
+
+    Checkout your dev branch with modified docs:
+    ```bash
+    git checkout [dev-branch]
+    ```
+
+    Generate API docs if necessary:
+    ```bash
+    python auto_doc.py
+    ```
+
+    Build docs with a version and alias
+    ```bash
+    mike deploy [version] [alias] --update-alias
+
+    # for example, if you branch is based on master and becomes new SNAPSHOT version:
+    mike deploy 2.2.0-SNAPSHOT dev --update-alias
```

Comment:
I have no clue what the version and alias are related to. The comment make it even stranger as you speak of master and `SNAPSHOT` but then have `2.2.0-SNAPSHOT` and `dev`. My brain then says that `2.2.0-SNAPSHOT` is for the `SNAPSHOT` part but this does not make sense as the order does not match the first part `based on master and becomes new SNAPSHOT`
I'm not sure it is clear as it is not clear for myself.
And if `2.2.0-SNAPSHOT` is for master this will not work once the master version get updated.

Reply:
okay I added a small section on how we actually map branches to versions on docs.hopsworks.ai. Please check if it make sit clearer.